### ISSUE TITLE
Multithreading based parallization

### ISSUE THREAD

**Issue by tjerkw** [cypress-io/cypress#3599]
Cypress parallelization only works on OS level (add more machines to add parallization). But there is no reason cypress couldn't run multiple tests in parallel on one machine, just like Selenium supports with the -w flag.

Ideally `cypress --parallelization=n` should just work.
It could start one headless browser for each threads (or process) and run thread in parallel.

Cypress tests are inherently IO bound (they are interacting with an application. 
Therefore it should be easy to scale UP (one big beafy multicore machine) instead of scalign out (multiple machine)

I know this goes against the payment model of cypress (pay for parallization). But as an open source tool we should have this supported.

**Comment by Falieson** [comment]
Also.... its really lame running my cypress suite on my MBP and have it take forever because its single threading the test files.

**Comment by Regenhardt** [comment]
Just now found out our quite strong CI runner can't run cypress tests in parallel 😞 
This seems to be an obvious feature.

**Comment by qalinn** [comment]
If this parallelization would be added maybe is a good idea to add a parameter where you can specify a list of the test environments against which you want to run the tests.
That means a real parallelization.

**Comment by million30** [comment]
Would this be included in the future release roadmap soon? Looks like the proposal has been accepted for quite a while.

**Comment by qalinn** [comment]
@million30 yes, It will be ice to have this implementation. Please take a look on my discussion/question. I have tried to explain how should looks this feature to have a big value. If it is not clear let me know and I will explain with more details.

please check here the discussion: https://github.com/cypress-io/cypress/discussions/15074#discussion-3219879

**Comment by SKFrozenCloud** [comment]
https://github.com/tnicola/cypress-parallel

Is that what you mean?

**Comment by qalinn** [comment]
@SKFrozenCloud Hello! no, is not the same. I want to be able to set baseurl for how many parallel tests you want to run.

**Comment by trainoasis** [comment]
Has someone found a workaround for this? We've recently implemented Cypress for e2e testing since a friend suggested it, but now coworkers started leaning towards Selenium just because of this feature. I'd prefer to stick with Cypress though, but seems this thread is not moving anywhere?
